mmr-lib/Randomizer.cs

```diff
@@ -219,68 +219,78 @@ private void EntranceShuffle()
 
         #region Sequences and BGM
 
-        private void BGMShuffle()
+        private void ShuffleBGM()
         {
-            while (RomData.TargetSequences.Count > 0)
+            var sequences = SequenceUtils.ReadSequenceInfo();
+
+            var targetSequences = new List<SequenceInfo>();
+
+            foreach (var item in sequences.Where(x => x.Replaceable))
             {
-                List<SequenceInfo> Unassigned = RomData.SequenceList.FindAll(u => u.Replaces == -1);
+                SequenceInfo targetSequence = new SequenceInfo
+                {
+                    Name = item.Name,
+                    Type = item.Type,
+                    Instrument = item.Instrument,
+                    Replaces = item.SequenceId
+                };
 
-                int targetIndex = Random.Next(RomData.TargetSequences.Count);
-                var targetSequence = RomData.TargetSequences[targetIndex];
+                targetSequences.Add(targetSequence);
+            }
+
+            sequences.RemoveAll(x => x.SequenceId == 0x18); //File Select, which is a duplicated entry
+
+            while (targetSequences.Count > 0)
+            {
+                List<SequenceInfo> Unassigned = sequences.FindAll(u => u.Replaces == -1);
+
+                int targetIndex = Random.Next(targetSequences.Count);
+                var targetSequence = targetSequences[targetIndex];
```

Comment:
I don't like this extension, primarily because i don't see choosing a random element in a list as a natural extension of a list object. It would make more sense for it to be an extension of the Random class. The secondary issue I have with it is just the name "Random" is too vague to explain what operation it's doing.

Reply:
I was going for making it similar to First, FirstOrDefault, Last, and LastOrDefault. Do you think it'd be better to be something like `var targetSequence = Random.GetElement(targetSequences);`?

Reply:
i think it's okish. I do know Python uses the more descriptive "choice" for a single pick one, and "choices" for a pick N (or i guess k, according to the docs)